packages/dds/tree/src/forest/cursorLegacy.ts

```diff
@@ -71,9 +71,10 @@ export interface ITreeCursor<TResult = TreeNavigationResult> {
     keys: Iterable<FieldKey>;
 
     /**
-     * @returns the number of immediate children for the given key of the currently selected node.
+     * @returns the number of immediate children for the given key of the currently selected node
+     * or the number of immediate children in the current field if the key is undefined.
      */
-    length(key: FieldKey): number;
+    length(key?: FieldKey): number;
```

Comment:
Rather than one method with two non-overlapping ways to use it, I'd prefer two methods.

This allows both methods to have simpler documentation and implementations (no more check the key and branch). This should be slightly faster (which is the main API design focus for cursor), as well as allow more specific names, types and documentation to improve clarity.

Having two separate methods also avoids the risk of a bug accidently trying to key a length of a field length with a possibly undefined key (ex: looking up a key in a map and passing it to length while forgetting to check for undefined: instead, you get a compile error in that case).

Maybe we could name the two methods currentFieldLength and childFieldLength.

Reply:
Done.